routes/deleteWarehouse/deleteWarehouse.js

```diff
@@ -0,0 +1,43 @@
+const express = require('express');
+const deleteWarehouse = express.Router();
+const fs = require('fs');
+
+//define variables
+let warehouses = [];
+let inventories = [];
+
+//middleware to fetch data from JSON file
+deleteWarehouse.use((req, res, next) => {
+  warehouses = [];
+  inventories = [];
+  //loads new data on to the arrays
+  warehouses = JSON.parse(fs.readFileSync('./data/warehouses.json'));
+  inventories = JSON.parse(fs.readFileSync('./data/inventories.json'));
+  next();
+});
+
+//DELETE request
+deleteWarehouse.delete('/:location', (req, res, next) => {
+  let warehouseToDelete = req.params.location;
+  //checks if requested warehouse exists on database
+  let warehouseCheck = warehouses.filter(
```

Comment:
So instead if alll the code from here to line 33 we could have simplified it in an easier way i know u r checking first if warehouse exists or not but if we use the find function we can just say if find function not return something then dont proceed else do some thing